src/firebolt/async_db/cursor.py

```diff
@@ -327,25 +338,31 @@ async def _api_request(
             content=query,
         )
 
+    async def _async_api_request(self, query: str) -> Response:
+        """Do query request using SET async_execution=1."""
+        return await self._api_request(
+            query, {"async_execution": 1, "advanced_mode": 1}
+        )
+
     async def _do_execute(
         self,
         raw_query: str,
         parameters: Sequence[Sequence[ParameterType]],
-        set_parameters: Optional[Dict] = None,
         skip_parsing: bool = False,
+        async_execution: Optional[bool] = False,
     ) -> None:
         self._reset()
-        if set_parameters is not None:
-            logger.warning(
-                "Passing set parameters as an argument is deprecated. Please run "
-                "a query 'SET <param> = <value>'"
+        if async_execution and "use_standard_sql" in self._set_parameters:
+            raise AsyncExecutionUnavailableError(
+                "It is not possible to execute queries asynchronously if "
+                "use_standard_sql is in use."
             )
         try:
 
             if parameters and skip_parsing:
                 logger.warning(
                     "Query formatting parameters are provided with skip_parsing."
-                    " They will be ignored"
+                    " They will be ignored."
                 )
 
             # Allow users to manually skip parsing for performance improvement
```

Comment:
We also need to error if user tries to execute a multistatement query. We need to error if `len(queries) > 0 and async_execution`

Reply:
A test will also be added. https://github.com/firebolt-db/firebolt-python-sdk/pull/187#discussion_r930235178